### PR TITLE
Homework Podstawy C++ #2

### PR DESCRIPTION
**Pull request by kontrybutor**
Authors: @kontrybutor, @PierwszyKrytykRP

### CODE REVIEW

File: module2/homework/nwdNww/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+  } else if (rhs == 0) {
+    return abs(lhs);
+  }
+  return NWD(rhs, lhs % rhs);
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+  return (lhs != 0 && rhs != 0) ? (abs(lhs) / NWD(lhs, rhs)) * abs(rhs) : 0;
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:15]: That's really on the edge of 'readable'.

Comment (kontrybutor) [reply module2/homework/nwdNww/nwdNww.hpp:15]: Thanks

Comment (ziobron) [reply module2/homework/nwdNww/nwdNww.hpp:15]: Yes, it's short but the main purpose of code is to be read by a human. The compiler will translate it to machine language, so we should always prefer readability over other things. Really rarely there are cases where unreadable code is introduced to have some optimisations.

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05
+3 XP cooperation

{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635, 1636],
  "points": [2, 4, 2, 6, 2, 5, 2, 5, 3],
  "users": ["kontrybutor", "PierwszyKrytykRP"]
}

**Pull request closed without merging**